src/client/Microsoft.Identity.Client/Internal/Broker/BrokerSilentRequest.cs

```diff
@@ -58,6 +58,12 @@ private async Task<MsalTokenResponse> SendAndVerifyResponseAsync()
 
         internal void CreateRequestParametersForBroker()
         {
+            if (string.IsNullOrEmpty(_silentParameters.LoginHint) && string.IsNullOrEmpty(_silentParameters.Account?.HomeAccountId?.Identifier))
```

Comment:
This should never happen - we should always have either a LoginHint or an IAccount. 
If the user decides to pass in their own object that implements IAccount (bummer), and their implemenation has HomeAccountId null, then we have to throw. Best to do that in SilentRequest.cs - there's a Validate method there.

Reply:
didnt see that method in silentrequest but it is in the acquireTokenSilentParameterBuilder. moved it there